lib/pool.dart

```diff
@@ -121,6 +121,44 @@ class Pool {
     });
   }
 
+  /// Requests a resource for the combined duration of the [callback] and the
+  /// processing of the Stream it returns.
+  /// (E.g. a database connection is hold until the the results get processed.)
+  Stream/*<T>*/ streamWithResource/*<T>*/(Stream/*<T>*/ callback()) {
+    StreamController/*<T>*/ streamController;
+    StreamSubscription/*<T>*/ streamSubscription;
+    Completer resourceCompleter = new Completer();
+    final complete = () {
+      if (streamSubscription != null) {
+        streamSubscription.cancel();
+        streamSubscription = null;
+      }
+      if (!resourceCompleter.isCompleted) {
+        resourceCompleter.complete();
+        streamController.close();
+      }
+    };
+    streamController = new StreamController/*<T>*/(onCancel: complete);
+    withResource(() {
+      if (resourceCompleter.isCompleted) return null;
+      new Future.sync(callback).then((Stream/*<T>*/ stream) {
+        if (stream == null) {
+          complete();
+          return null;
+        }
+        streamSubscription = stream.listen(streamController.add,
+            onError: streamController.addError,
+            onDone: complete,
+            cancelOnError: true);
```

Comment:
It looks like the updated tests don't use `FakeAsync` anymore. Why is that? Could the failures be caused by a race condition?

If you update the PR with the failing tests, I can patch it in and see what I can figure out.